### DISCUSSION TITLE
Improve the look of reply buttons.

### DISCUSSION THREAD

**Discussion by occluder** [Chatterino/chatterino2#4185] [Ideas]
With the introduction of replies in Chatterino, an option to add a visible reply button near each message has been added. 

This option is from what I have gathered rarely enabled, most agree that the buttons look very unfitting and out-of-place on Chatterino. Consequently, I wanted to have a suggestion open about this matter. 

Some ideas for improvements could be:

- Making the buttons look & behave closer to Discord and Twitch, by making it show a button top-right of the message being hovered over:

    <img src="https://user-images.githubusercontent.com/69414142/203586170-ecc21e1f-34c5-4ab4-bea8-34c83b2d8294.png" width="450px" height="100px">

    ![image](https://user-images.githubusercontent.com/69414142/203586370-4521c0ac-7f55-449a-8cea-bb3c4efa99d4.png)

- Giving the buttons a lower visibility when they are not being hovered over.
- Changing the button's image to blend in better with the background, while staying reasonably visible.

I understand that UI/UX isn't a top priority for Chatterino, and that it will take time for this to be considered/revised. So of course this is an open place for discussion; If others have cool ideas to make the buttons look better, feel free to drop them in the replies!

**Comment by x9136** [comment]
Showing the reply button on hovering is great because it reduces chat clutter.

**Reply by Felanbird** [reply]
There's a section in the original PR explaining why the button isn't on-hover
https://github.com/Chatterino/chatterino2/pull/3722

**Reply by x9136** [reply]
Got it.

What about the location? Messages often have different lengths, so the button at the end of the message will always be in different locations. Because of this you always have to aim, and visually the buttons make a snake.

If you put the button to the left of the subscription icons and others, it is easier to aim, there is less visual noise in the message area. The button will stand next to similar functions that happen when you click on a nickname.

![x](https://github.com/Chatterino/chatterino2/assets/60105060/f01d35df-55a4-4cd1-8dc7-f88fb80ad8e9)

**Comment by x9136** [comment]
[What is the difficulty?](https://github.com/Chatterino/chatterino2/discussions/4185#discussioncomment-5886787)

**Reply by Nerixyz** [reply]
The difficulty is doing this while maintaining an acceptable performance and fitting into the current rendering infrastructure. Currently, there's no easy support for overlay widgets.

**Reply by x9136** [reply]
> Currently, there's no easy support for overlay widgets.

I do not know if you see the picture above, because my link leads to it. If not, there is a simple solution — the buttons have been moved to another location. That's what I asked about, because the question about the overlay was answered a year ago.

**Reply by Nerixyz** [reply]
My bad, GitHub didn't really highlight the comment. That would be possible, but I think the issue is that the old way of doing things has been established. We could try to have some kind of hover effect (there's the go-to-message overlay), but I feel like it would be a rather large hack.

**Reply by x9136** [reply]
Although the situation with the reply button can be improved by moving, I realized that the general approach is this: everything will remain as it is until there is full-fledged hover support or until someone writes their own hover component.
